src/Symfony/Component/Decorator/README.md

```diff
@@ -0,0 +1,72 @@
+Clock Component
+===============
+
+Symfony Decorator modify the functionality of other functions or methods.
+
+**This Component is experimental**.
+[Experimental features](https://symfony.com/doc/current/contributing/code/experimental.html)
+are not covered by Symfony's
+[Backward Compatibility Promise](https://symfony.com/doc/current/contributing/code/bc.html).
+
+Getting Started
+---------------
+
+```bash
+composer require symfony/decorator
+```
+
+```php
+use Symfony\Component\Decorator\Attribute\Decorate;
+use Symfony\Component\Decorator\DecoratorChain;
+use Symfony\Component\Decorator\DecoratorInterface;
+
+class DebugDecorator implements DecoratorInterface
+{
+    public function decorate(\Closure $func): \Closure
+    {
+        return function (mixed ...$args) use ($func): mixed {
+            echo "Do something before\n";
+
+            $result = $func(...$args);
+
+            echo "Do something after\n";
+
+            return $result;
+        }
+    }
+}
+
+$decorator = DecoratorChain::from([
+    DebugDecorator::class => fn () => new DebugDecorator(),
+]);
+
+#[\Attribute(\Attribute::TARGET_FUNCTION | \Attribute::TARGET_METHOD | \Attribute::IS_REPEATABLE)]
+class Debug extends Decorate
+{
+    public function __construct()
+    {
+        parent::__construct(DebugDecorator::class);
+    }
+}
+
+class Greeting
+{
+    #[Debug]
+    public function hello(string $name): void
+    {
+        echo "Hello $name\n"
+    }
+}
+
+$greeting = new Greeting();
+$decorator->call($greeting->hello(...), 'Fabien');
```

Comment:
Perhaps write the output to give full example?

Reply:
Output added, thanks!